### PR TITLE
Improve Command infrastructure

### PR DESCRIPTION
**Pull request by mdekrey**
Improves in several ways:
- Composition over inheritance
- Covariance/contravariance
- Separate command framework
- Improve compliance with Interface Segregation Principle

This also leverages the `GlobalUsings.cs` to have type aliases.

Also, the `GitCloneService` was written to very carefully not call any `IGitToolsCommand` instances, which required a git clone to work; by requiring a separate command invoker for non-git-tools-commands, the `GitCloneService` is no longer fragile in this way.

### CODE REVIEW

**Review by chelfersps — APPROVED** [review]
💪

**Review by gitpo — APPROVED** [review]

File: Server/Commands/DisposableContextCommandInvoker.cs [hunk 1]
```diff
@@ -0,0 +1,14 @@
+﻿namespace PrincipleStudios.ScaledGitApp.Commands;
+
+public class DisposableContextCommandInvoker<TContext>(Func<Task<DisposableContext<TContext>>> constructor, ILogger logger) : ICommandInvoker<TContext>
+{
+	public async Task RunCommand(ICommand<Task, TContext> command) =>
+		await await RunGenericCommand(command);
```
Comment (Sunnova-PhillipJacobs) [inline Server/Commands/DisposableContextCommandInvoker.cs:6]: await await?

Comment (pcj02b) [inline Server/Commands/DisposableContextCommandInvoker.cs:6]: await await?

Comment (pcj02b) [reply Server/Commands/DisposableContextCommandInvoker.cs:6]: oh. It is actually await await.

**Pull request merged**